packages/fe-fpm-writer/src/building-block/index.ts

```diff
@@ -0,0 +1,155 @@
+import { render } from 'ejs';
+import type { Editor } from 'mem-fs-editor';
+import { join } from 'path';
+import type { BuildingBlock } from './types';
+import { DOMParser, XMLSerializer } from '@xmldom/xmldom';
+import * as xpath from 'xpath';
+import format from 'xml-formatter';
+import { getErrorMessage, validateBasePath } from '../common/validate';
+
+/**
+ * Generates a building block into the provided xml view file.
+ *
+ * @param {string} basePath - the base path
+ * @param {string} viewOrFragmentPath - the path of the view or fragment xml file relative to the base path
+ * @param {string} aggregationPath - the aggregation xpath
+ * @param {BuildingBlock} buildingBlockData - the building block data
+ * @param {Editor} [fs] - the memfs editor instance
+ * @returns {Editor} the updated memfs editor instance
+ */
+export function generateBuildingBlock<T extends BuildingBlock>(
```

Comment:
Sorry, I missed this in my first review, however, i would like to have a the `generate` functions consistent in the module. We always have three parameters: `basePath`, an object will all specific configs, and optional `fs`.

Reply:
https://github.com/SAP/open-ux-tools/pull/555/commits/9229ec75d8eff92ceeca032f392eda1c602d6b5a